internal/pkg/plugin/jenkins/jenkins.go

```diff
@@ -29,20 +32,46 @@ func genJenkinsState(options plugininstaller.RawOptions) (statemanager.ResourceS
 		return nil, err
 	}
 
-	// svc_name.svc_ns:svc_port
-	outputs := map[string]interface{}{
-		"jenkins_url": "http://jenkins.jenkins:8080",
-	}
-	resState.SetOutputs(outputs)
-
 	// values.yaml
 	opt, err := helm.NewOptions(options)
 	if err != nil {
 		return nil, err
 	}
-
 	valuesYaml := opt.GetHelmParam().Chart.ValuesYaml
 	resState["values_yaml"] = valuesYaml
 
+	svcName, err := genJenkinsSvcName(options)
+	if err != nil {
+		return nil, err
+	}
+
+	// svc_name.svc_ns:svc_port
+	url := fmt.Sprintf("http://%s.%s:8080", svcName, opt.Chart.Namespace)
+	outputs := map[string]interface{}{
+		"jenkins_url": url,
+	}
+
+	resState.SetOutputs(outputs)
+
 	return resState, nil
 }
+
+// see https://github.com/devstream-io/devstream/pull/1025#discussion_r952277174 for more info.
+func genJenkinsSvcName(options plugininstaller.RawOptions) (string, error) {
+	opts, err := helm.NewOptions(options)
+	if err != nil {
+		return "", err
+	}
+
+	var retStr string
+	if strings.Contains(opts.Chart.ChartName, opts.Chart.ReleaseName) {
+		retStr = strings.TrimSuffix(opts.Chart.ReleaseName, "-")
+		if len(retStr) > 63 {
+			retStr = retStr[:64]
+		}
+		return retStr, nil
+	}
+	retStr = fmt.Sprintf("%s-%s", opts.Chart.ReleaseName, opts.Chart.ChartName)
+
```

Comment:
![image](https://user-images.githubusercontent.com/61452000/186350603-7182b4fb-7f16-46c4-a908-79dfbaf1cefe.png)

From the pic, the `if` and `else` branches all require `trunc 63` and `trimSuffix`.

Reply:
or may be written like this:
```go
pipe := func(s string) string {
	if len(s) > 63 {
		s = s[:64]
	}
	return strings.TrimSuffix(s, "-")
}

if strings.Contains(opts.Chart.ChartName, opts.Chart.ReleaseName) {
	return pipe(opts.Chart.ReleaseName), nil
}
return pipe(fmt.Sprintf("%s-%s", opts.Chart.ReleaseName, opts.Chart.ChartName)), nil
```

Reply:
Good tips. But I think the original requirements are:

1. length <=63
2. suffix != "-"

So trim suffix first is better.

Reply:
But if the last character after you trim is `-`

Reply:
`hell- world`
1. trunc 5 | trim => `hell`
2. trim | trunc 5 => `hell-`

Reply:
Yeah, I agree with you. I'll fix it asap.